pymemcache/client/retrying.py

```diff
@@ -0,0 +1,297 @@
+""" Module containing the RetryingClient wrapper class. """
+
+from collections import Iterable
+from time import sleep
+
+
+def _ensure_tuple_argument(argument_name, argument):
+    """
+    Helper function to ensure the given arguments are tuples of Exceptions (or subclasses),
+    or can at least be converted to such.
+
+    Args:
+      argument_name: str, name of the argument we're checking, only used for raising meaningfull
+        exceptions.
+      argument: any, the argument itself.
+
+    Returns:
+      tuple[Exception]: A tuple with the elements from the argument if they are valid.
+
+    Exceptions:
+      ValueError: If the argument was not None, tuple or Iterable.
+      ValueError: If any of the elements of the argument is not a subclass of Exception.
+    """
+
+    argument_tuple = None
+
+    # Ensure the argument is a tuple, or raise an error if we can't convert it.
+    if argument is None:
+        return tuple()
+    elif isinstance(argument, tuple):
+        argument_tuple = argument
+    elif isinstance(argument, Iterable) and not isinstance(argument, (str, bytes, bytearray)):
+        argument_tuple = tuple(argument)
+    else:
+        raise ValueError("%s was not a tuple or Iterable." % argument_name)
+
+    # Check that all the elements are actually inherited from Exception. (Catchable)
+    if not all([issubclass(arg, Exception) for arg in argument_tuple]):
+        raise ValueError("%s contained elements other than Exceptions." % argument_name)
+
+class RetryingClient(object):
+    """
+    Client that allows retrying calls for the other clients.
+    """
+
+    def __init__(
```

Comment:
Document the arguments to this method. For example, what does it mean for `retry_for` to be `None`? Does it mean that all exceptions cause retries, or no exceptions cause retries?